app/services/courses/assign_program_type_service.rb

```diff
@@ -17,8 +17,6 @@ def execute(funding_type, course)
       when 'fee'
         course.program_type = calculate_fee_program(course)
       end
-      # NOTE: This looks like unwarranted side effects
-      course.save
     end
```

Comment:
This is a hangover from a previous version of the application.

This service gets called during the building of a new course. When `course.save` is called, `validate` is called on the course, and then all custom errors added to the model are cleared and only the validations defined on the model are run.

This is not what we want when incrementally building up errors in the new course flow.